rtos/Mutex.h

```diff
@@ -73,23 +73,39 @@ class Mutex : private mbed::NonCopyable<Mutex> {
     /** Create and Initialize a Mutex object
 
      @param name name to be used for this mutex. It has to stay allocated for the lifetime of the thread.
-
      @note You cannot call this function from ISR context.
     */
     Mutex(const char *name);
 
-    /** Wait until a Mutex becomes available.
-      @param   millisec  timeout value or 0 in case of no time-out. (default: osWaitForever)
+    /**
+      Wait until a Mutex becomes available.
+
+      @return  status code that indicates the execution status of the function:
+               @a osOK the mutex has been obtained.
+
+      @note You cannot call this function from ISR context.
+      @note This function treats RTOS errors as fatal system errors, so can only return osOK.
+            Use of the return value is deprecated, as the return is expected to become void in the future.
+     */
+    osStatus lock(void);
+
+    /**
+      For backwards compatibility.
+      @deprecated Do not use this function. This function has been replaced with lock(), trylock() and trylock_for() functions.
+
+      Wait until a Mutex becomes available.
+      @param   millisec  timeout value or 0 in case of no time-out.
       @return  status code that indicates the execution status of the function:
                @a osOK the mutex has been obtained.
                @a osErrorTimeout the mutex could not be obtained in the given time.
-               @a osErrorParameter internal error.
                @a osErrorResource the mutex could not be obtained when no timeout was specified.
-               @a osErrorISR this function cannot be called from the interrupt service routine.
 
       @note You cannot call this function from ISR context.
+      @note This function treats RTOS errors as fatal system errors, so can only return osOK or
+            osErrorResource in case when millisec is 0 or osErrorTimeout if millisec is not osWaitForever.
      */
-    osStatus lock(uint32_t millisec=osWaitForever);
+    MBED_DEPRECATED_SINCE("mbed-os-5.10.0", "Replaced with lock(), trylock() and trylock_for() functions")
```

Comment:
Shouldn't be here `;` ? Look at exporters failing (this line there, as `mbed-os\rtos/Mutex.h(107): error:  #79: expected a type specifier`)